rllib/metrics/metric.py

```diff
@@ -0,0 +1,88 @@
+from abc import abstractmethod
+from typing import Dict
+
+
+class Metric:
+    """ Abstact class for metric callbacks. Please see MetricWrapper
+    for more details """
+
+    @staticmethod
+    @abstractmethod
+    def on_episode_start(info: Dict) -> None:
+        """
+        Runs before an episode starts. This is where buffer initialization and
+        counter resetting is being done.info["episode"].user_data is the
+        natural place to store intermediate data
+        :param info: dictionary containing the following keys:
+            "env": ray.rllib.BaseEnv instance running the episodes
+            "policy": Dict[str, Policy] - mapping all available policies
+            "episode": MultiAgentEpisode - see class
```

Comment:
These docstrings are very useful. Should we lift them to top level params?, e.g.
```
def on_episode_start(env: BaseEnv, policy: Dict[str, Policy], episode: MultiAgentEpisode):
    """Runs before an episode starts.

    Args:
        env: ray.rllib.BaseEnv instance running the episodes.
        policy: Dict[str, Policy] - mapping all available policies.
        episode: MultiAgentEpisode - the current episode state.
    """